class.jetpack.php

```diff
@@ -378,10 +378,14 @@ function plugin_upgrade() {
 			}
 
 			//if Jetpack is connected check if jetpack_unique_connection exists and if not then set it
-			if ( ! get_option( 'jetpack_unique_connection' ) ) {
+			$jetpack_unique_connection = get_option( 'jetpack_unique_connection' );
+			if ( $jetpack_unique_connection && array_key_exists( 'version', $jetpack_unique_connection ) ) {
+				return;
```

Comment:
I don't think we should return here because there is more code that is supposed to run after this conditional